pyccel/ast/core.py

```diff
@@ -5177,15 +5177,56 @@ def bodies(self):
         return b
 
 
-class IfTernaryOperator(If):
+class IfTernaryOperator(Basic, PyccelAstNode):
+    """Represents a IfTernaryOperator statement in the code.
 
-    """class for the Ternery operator"""
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : [condition , value_if_true, value_if_false]
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign, IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    """
     def __init__(self, *args):
-        for arg in self.args:
-            if len(arg[1].body)!=1:
-                raise TypeError('IfTernary body must be of length 1')
+        self._args = []
+        cond = args[0]
+        first = args[1]
+        second = args[2]
+        self._args = [cond, first, second]
+        if self.stage == 'syntactic':
+            return
 
-    pass
+        if isinstance(first , Nil) or isinstance(second, Nil):
+            raise TypeError('None is not implemeted for Ternary Operator')
+        if first.shape != second.shape :
+            raise TypeError('results in Ternary Operator should be the same shape')
+        if isinstance(first.dtype, NativeString) ^ isinstance(second.dtype, NativeString):
+            raise TypeError('Only one of the Ternary Operator results is type string')
+        _tmp_list = [NativeBool(), NativeInteger(), NativeReal(), NativeComplex(), NativeString()]
+        if first.dtype not in _tmp_list :
+            raise TypeError('cannot determine the type of {}'.format(first.dtype))
+        if second.dtype not in _tmp_list :
+            raise TypeError('cannot determine the type of {}'.format(second.dtype))
+        self._dtype = max([first.dtype, second.dtype], key = lambda x : _tmp_list.index(x))
+        self._precision = max([first.precision, second.precision])
+        if None in [first.rank, second.rank]:
+            self._rank = None
+            self.shape = None
+        else :
+            self._shape = broadcast(first.shape, second.shape)
```

Comment:
It is interesting that you chose to use `broadcast` and yet you require the shape of `first` and `second` to be the same. Could this restriction be relaxed?

Reply:
True, I didn't pay attention to _broadcast__  function. checking shape is not needed.